### ISSUE TITLE
The screenshot text annotation function conflicts with the Chinese Pinyin input method.| 截图文字标注功能与中文拼音输入法冲突

### ISSUE THREAD

**Issue by geekyouth** [ShareX/ShareX#3621]
When you take a screenshot, you can use the Chinese Pinyin input method, type Pinyin, and press enter to submit the text. This will result in direct submission of the entire screenshot, instead of just submitting Pinyin input method candidates, which is very bad.
***
截图时，标注文字弹窗界面，使用中文拼音输入法，键入拼音，按下回车键 提交文字过程中，会导致提直接交整个截图，而不是仅提交拼音输入法候选文字，这很糟糕。
***
![image](https://user-images.githubusercontent.com/12899262/45467147-97644e00-b750-11e8-9307-e3c06744c7c9.png)

**Comment by Scrxtchy** [comment]
Can easily replicate this issue without relying on a CJK input
Using `Win+;` or `Win+.` for emoji input on Windows 10. The Enter key is also used for commiting your emoji selection to the text area
In sharex, this in addition commits the text to the text box on the canvas

Ctrl+Enter should be the button to commit text, like it's used forums and github comments

**Comment by Jaex** [comment]
I don't prefer to use ctrl + enter. Most chat clients using just enter to send message how you guys using pinyin there then?

**Comment by Scrxtchy** [comment]
Because pressing enter with punyin input activated doesn't send the message. Only sharex does this

**Comment by tinybarks** [comment]
I think adding a configuration option that disables the Enter key might work, or maybe we can detect if an IME is being used in a press-enter-to-pick mode ([it seems you can check for this](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control.imemode?view=netframework-4.7.2#System_Windows_Forms_Control_ImeMode)) and disable the enter hotkey in that situation.

**Comment by Scrxtchy** [comment]
This is for setting the IME setting for a control on winforms.
A similar function to how firefox sets your input language to alphanumeric, but windows knows your currently using katakana, so it switches it back when you return to something like notepad
![18-10-04_18-39-44](https://user-images.githubusercontent.com/12771982/46462497-f45d9c00-c804-11e8-9dad-2509559510a7.gif)

**Comment by tinybarks** [comment]
It can be used to set the IME mode or to disable IME, but you missed an important note in the docs:

>  The user can *interactively change* the Control.ImeMode unless it has been programmatically set to Disable. In this case, the user will not be able to use the IME. 

It seems to work in testing, but it's not good enough (only detects IME settings if you set it to a default value like `Off` which is hacky and apparently doesn't work with global input). I'll experiment with [this](https://docs.microsoft.com/en-us/windows/desktop/api/imm/nf-imm-immgetconversionstatus) later.

**Comment by tinybarks** [comment]
`ImmGetConversionStatus()` doesn't seem to work for Microsoft Pinyin, unfortunately. `ImeMode` does seem to give reliable indication of IME status, but only if you change `ImeMode` to a different value than `NoControl` first... and that's hacky and bad (see comment above).

I think what may work is checking whether `InputLanguage.CurrentInputLanguage.Culture.TwoLetterISOLanguageName` is one of "zh", "ja" or "ko"
 where the <kbd>Enter</kbd> keypress is handled. If that condition is met, the user is probably using a keyboard layout with an IME and we should disable commit on <kbd>Enter</kbd> so IME input can be used.

**Comment by Scrxtchy** [comment]
How about when you input emoji when on english?
https://tenryuu.blob.core.windows.net/astrid/2018/10/18-10-07_00-56-51.webm

**Comment by tinybarks** [comment]
The thing I just committed is some code I tested with Google Japanese Input, Microsoft Japanese IME, and Microsoft Pinyin on Windows 10 1803. This should distinguish between <kbd>Enter</kbd> pressed in the suggestions mode and <kbd>Enter</kbd> pressed in other contexts, so it'll behave as you expect.

I'm not sure if this works with other IMEs, but it seems like all the ones I tested will, when entering things in suggestions mode, continue to pass key events to the field but replace the `KeyDown` event and, depending on the IME, the `KeyUp` event `KeyCode` with `VK_PROCESSKEY`. 

For example, entering "キツネ" on the Microsoft Japanese IME, then pressing enter looks like:

```
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: K, kc: K, kv: 75
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: I, kc: I, kv: 73
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: T, kc: T, kv: 84
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: S, kc: S, kv: 83
keyup kd: U, kc: U, kv: 85
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: N, kc: N, kv: 78
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: E, kc: E, kv: 69
keydown kd: ProcessKey, kc: ProcessKey, kv: 229
keyup kd: Return, kc: Return, kv: 13
keydown kd: Return, kc: Return, kv: 13 
keyup kd: Return, kc: Return, kv: 13
```

Typing anything while in Microsoft Pinyin suggestions just gives you `VK_PROCESSKEY` messages until suggestion mode is left, so it's similar. 

@Scrxtchy, I tried this code with the emoji input and it didn't work. But the emoji input doesn't seem to work like an IME does. Instead, when you use it with a text box, you get exactly the key presses the user performs, and when the user presses enter, it seems to send three `KeyPress` events, adds the first character of the surrogate pair to the `Text` property, sends one more `KeyPress` event, then adds the second character to `Text`, then fires a `KeyUp` event with kc = `Return` (but no `KeyDown` event). That seems way more annoying to track.
